src/keymgmt.c

```diff
@@ -498,9 +499,13 @@ static int p11prov_common_gen(struct key_generator *ctx,
     sh = p11prov_session_handle(session);
 
     if (cka_id.ulValueLen == 0) {
+        int err = RET_OSSL_ERR;
         /* generate unique id for the key */
-        ret = p11prov_GenerateRandom(ctx->provctx, sh, id, sizeof(id));
-        if (ret != CKR_OK) {
+        err = RAND_priv_bytes_ex(p11prov_ctx_get_libctx(ctx->provctx), id,
+                                 sizeof(id), 0);
```

Comment:
Why do we need priv_bytes here?

Reply:
Uhmm indeed we do not actually need that .. let me change it to the public one.